### PR TITLE
d4xx: return hwmc error code to lrs sdk.

### PR DESCRIPTION
**Pull request by dmipx**
Align MIPI driver NOT_READY error to UVC flow

Tracked-by: [RSDSO-19622]

### CODE REVIEW

File: kernel/realsense/d4xx.c [hunk 1]
```diff
@@ -5639,4 +5628,4 @@ MODULE_AUTHOR("Guennadi Liakhovetski <guennadi.liakhovetski@intel.com>,\n\
 				Shikun Ding <shikun.ding@intel.com>");
 MODULE_AUTHOR("Dmitry Perchanov <dmitry.perchanov@intel.com>");
 MODULE_LICENSE("GPL v2");
-MODULE_VERSION("1.0.1.20");
+MODULE_VERSION("1.0.1.21");
```
Comment (Nir-Az) [inline kernel/realsense/d4xx.c:5631]: Please bump to 22 as 5.16 RC is 21
https://github.com/IntelRealSense/realsense_mipi_platform_driver/blob/db6472e0590596a33c37bca9834ca37833aceefb/kernel/realsense/d4xx.c#L5642

**Comment by Nir-Az** [comment]
We must add live tests gating to the driver, hard to tell if this change doesn't break anything else..

File: kernel/realsense/d4xx.c (hunk 1 above)
Comment (Nir-Az) [reply kernel/realsense/d4xx.c:5631]: Please remove this change as the version was already updated

Comment (dmipx) [reply kernel/realsense/d4xx.c]: squash and drop unnecessary commits.

**Comment by dmipx** [comment]
> We must add live tests gating to the driver, hard to tell if this change doesn't break anything else..

Good idea. I will add unit-test

**Pull request merged**